openedx/core/djangoapps/agreements/views.py

```diff
@@ -0,0 +1,93 @@
+"""Views served by the honor code app. """
+
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from opaque_keys.edx.keys import CourseKey
+
+from common.djangoapps.student import auth
+from common.djangoapps.student.roles import CourseStaffRole
+
+from .api import get_integrity_signature
+
+
+def is_user_course_or_global_staff(user, course_id):
```

Comment:
#NIT I enjoyed having this as a decorator. Do you think that's a good pattern?

Reply:
I don't want to have it as a decorator on the function because we still want students to have access to the endpoint